### PR TITLE
Switch flutter_tools from script to app-jit snapshot.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Small startup improvement to tools time as measured on a mac:

script:
```
 time flutter > /dev/null

real	0m0.549s
user	0m0.416s
sys	0m0.206s
```

app-jit 
```
time flutter > /dev/null

real	0m0.326s
user	0m0.251s
sys	0m0.178s
```

Also added a training command where we could attempt to exercise more of the tool to get better training. Requires disabling asserts for certain tools tests

### CODE REVIEW

**Comment by jonahwilliams** [comment]
These tests do not fail locally, investigating as a potential issue with the cirrus env

File: dev/bots/test.dart [hunk 1]
```diff
@@ -181,7 +181,7 @@ Future<void> _runToolTests() async {
   await _buildRunnerTest(
     path.join(flutterRoot, 'packages', 'flutter_tools'),
     flutterRoot,
-    enableFlutterToolAsserts: true,
+    enableFlutterToolAsserts: false, // required to use app-jit tool snapshots.
```
Comment (yjbanov) [inline dev/bots/test.dart]: Would it be possible to ever flip this flag to `true`? If not, maybe remove it?

File: packages/flutter_tools/lib/src/commands/train.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../runner/flutter_command.dart';
+
+// This command is run when generating the app-jit snapshot for the tool, so it cannot access the Cache
+// or any artifacts that haven't been downloaded yet.
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/train.dart]: nit: `///`

File: packages/flutter_tools/lib/src/commands/train.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+class TrainingCommand extends FlutterCommand {
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/train.dart]: There's not a lot of training here :)

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

**Comment by jonahwilliams** [comment]
The tool_tests-linux has actually passed, submitting

**Comment by jonahwilliams** [comment]
This had a small, but measurable speedup on flutter test - yay!

**Comment by yjbanov** [comment]
:fireworks:

**Comment by yjbanov** [comment]
Let's update the baseline!

**Pull request merged**